src/net.cpp

```diff
@@ -1031,6 +1051,9 @@ void ThreadSocketHandler()
             BOOST_FOREACH(CNode* pnode, vNodesCopy)
                 pnode->Release();
         }
+
+        if (progress == 0)  // Nothing happened even though select did not block.  So slow us down.
+            MilliSleep(50);
```

Comment:
50 seems a bit arbitrary. This looks like it could busy wait a lot if the bandwidth throttler is set really low. Does it matter in practice? Maybe it should try and guess a good sleep value based on the throttle settings?